scripts/deploy-tokenizedBallot.ts

```diff
@@ -0,0 +1,45 @@
+import 'dotenv/config';
+import {ethers} from 'hardhat';
+import {TokenizedBallot__factory} from '../typechain-types';
+
+export const deployTokenizedBallot = async (): Promise<void> => {
+  const {ALCHEMY_API_KEY} = process.env;
+  const GOERLI_PRIVATE_KEY = process.env.GOERLI_PRIVATE_KEY as string; // getting linter issues if not assigning type.
+  const [networkName, myTokenAddress, ...proposals] = process.argv.slice(3);
+
+  if (!proposals || proposals.length === 0) {
+    throw new Error('You need to pass a valid constructor argument');
+  }
+  const PROPOSALS_BYTES32 = proposals.map(p => ethers.utils.formatBytes32String(p));
+
+  try {
+    let ContractFactory: TokenizedBallot__factory;
+    let blockNumber: number;
+    if (networkName === 'goerli') {
+      if (!ALCHEMY_API_KEY) {
+        throw new Error('ALCHEMY_API_KEY missing');
+      }
+      const provider = new ethers.providers.AlchemyProvider('goerli', ALCHEMY_API_KEY);
+      const wallet = new ethers.Wallet(GOERLI_PRIVATE_KEY, provider);
+      const signer = wallet.connect(provider);
+      ContractFactory = new TokenizedBallot__factory(signer);
+      const lastBlock = await ethers.provider.getBlock('latest');
+      blockNumber = lastBlock.number - 1; // set this is argument instead? use a blocknumber 12 hours in the future?
```

Comment:
perhaps better to align the name with what it's called in the contract - `targetBlockNumber`

Reply:
You could also completely remove this arg and just get the block number in the contract constructor at time of deployment. idk what the intended behavior is though

Reply:
Yes, I considered this as well as you can see by comment on line 28.